server/ntb/io/feeding_services/ntb_reuters_api.py

```diff
@@ -0,0 +1,226 @@
+
+    ERRORS = [
+        IngestApiError.apiTimeoutError().get_error_description(),
+        IngestApiError.apiRedirectError().get_error_description(),
+        IngestApiError.apiRequestError().get_error_description(),
+        IngestApiError.apiUnicodeError().get_error_description(),
+        IngestApiError.apiParseError().get_error_description(),
+        IngestApiError.apiGeneralError().get_error_description(),
+    ]
+
+    label = "NTB Reuters feed API"
+
+    fields = [
+        {
+            "id": "client_id",
+            "type": "password",
+            "label": "client_id",
+            "placeholder": "Client Id",
+            "required": True,
+        },
+        {
+            "id": "client_secret",
+            "type": "password",
+            "label": "client_secret",
+            "placeholder": "Client Secret",
+            "required": True,
+        },
+        {
+            "id": "audience",
+            "type": "text",
+            "label": "audience",
+            "placeholder": "Audience",
+            "required": True,
+        },
+        {
+            "id": "topic",
+            "type": "text",
+            "label": "topic",
+            "placeholder": "Topic",
+            "required": False,
+        },
+        {
+            "id": "channel",
+            "type": "text",
+            "label": "channel",
+            "placeholder": "channel",
+            "required": False,
+        },
+    ]
+
+    session = None
+
+    def _update(self, provider, update):
+        items = []
+        self.provider = provider
+        self.session = requests.Session()
+        provider_config = self.provider.get("config")
+
+        if not provider_config.get("token") or self.is_token_expired(provider_config):
+            self.auth(provider, provider_config)
+
+        headers = {
+            "Authorization": f'Bearer {provider_config.get("token")}',
+            "Content-Type": "application/json",
+        }
+
+        cursor = ""
+
+        while True:
+            try:
+                variables = {
+                    "channel": provider_config.get("channel", ""),
+                    "topicCodes": provider_config.get("topic", ""),
+                    "cursor": cursor,
+                    "dateRange": provider.get(
+                        "last_updated", datetime.datetime.now
+                    ).strftime("%Y.%m.%d"),
+                }
+                response = self.session.post(
+                    provider_config.get("url"),
+                    headers=headers,
+                    data=json.dumps(
+                        {"query": self.get_query(), "variables": variables}
+                    ),
+                    timeout=30,
+                )
+                response.raise_for_status()
+                data = response.json()
+
+            except requests.exceptions.HTTPError as e:
+                logger.error(e)
+                return
+
+            parser = self.get_feed_parser(provider)
```

Comment:
not sure if we're creating a new instance every time, but in case we do it would be better to do this once before the loop